### DISCUSSION TITLE
Using SEQ on Azure VM

### DISCUSSION THREAD

**Discussion by beaudamore** [datalust/seq-tickets#1858] [Help]
I have an azure vm that refuses to accept (ingest) anything from any other ip except localhost.

For "listenUris" I have tried:

- the public IP:5341,the Azure DNS name for the VM:5341
- the public IP:80,the Azure DNS name for the VM:5341
- the public IP:5341,the Azure DNS name for the VM:45341 (and setting 'ingestionPort' to 45341)

..and every swappable combo I haven't listed. Most of the time, anything other than localhost or localhost:5341 results in not being able to pull up the UI.

After looking through the SEQ logs showing me connection errors it **_recommended_** I also run some of these commands:

`netsh http add urlacl url=http://<publicIP>:5341/ user=<VMname>\\SeqDefaultInstance `
`netsh http add urlacl url=http://<AzureDNSNAme>:5341/ user=<VMname>\\SeqDefaultInstance `

I wound up doing all inclusive commands as such:

`netsh http add urlacl url=http://+:80/ user=<VMname>\\SeqDefaultInstance  `
`netsh http add urlacl url=http://+:5341/ user=<VMname>\\SeqDefaultInstance  `

However, now I am getting different errors:
`"Error running the server application","@l":"Fatal","@x":"Microsoft.AspNetCore.Server.HttpSys.HttpSysException (1214): The format of the specified network name is invalid.\r\n `

I have confirmed the ports are open by being able to hit the SEQ Web UI at :5341 and :80 (and of course 3389 for RD) but this is all I get from running test logs with:

`seqcli log -m "Test" -s http://<AzurePublicIP>:5341`
`seqcli log -m "Test" -s http://<AzureDNSName>:5341`

"A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. "

The only test that works is:

`seqcli log -m "Test" -s http://localhost:5341 `

on the VM No outside test makes it in...
I am at an impass now. 

Thanks for reading.

**Comment by nblumhardt** [comment]
Hi Beau, thanks for your message.

At first glance this doesn't seem right to me; the Seq UI uses the exact same HTTP endpoints as `seqcli` would. My best guess, then, is that perhaps your web browser is respecting some kind of network or proxy settings that `seqcli` isn't 🤔 

Could that be the case, or does it spark any ideas?

**Comment by beaudamore** [comment]
I wound up reverting back to:
"http://localhost:5341/" for the listenUrls value only
and
netsh http add urlacl url=http://publicIP:5341/ user=<VMname>\\SeqDefaultInstance  
netsh http add urlacl url=http://publicIP:80/ user=<VMname>\\SeqDefaultInstance  
and
for the SEQ appender in .NET, I use the AzureDNSName... and it works 
BUT get this... it works from the other app services but still not from itself or externally outside azure...?

I cannot push using seqcli log even from itself using anything other than localhost...

At this point, I am glad it's working even though it doesn't make sense...

**Reply by yooakim** [reply]
This could be because of network security groups. Often all Azure Services are allowed but not VMs. Can you check if the VM has network rules blocking access from outside of Azure?

**Comment by beaudamore** [comment]
No rules blocking. It has RDP, http, https access as well.

**Reply by yooakim** [reply]
Just a follow up, where is the SSL connection terminated? (who manages the SSL connection?)

**Comment by beaudamore** [comment]
I manage it. It is a lone vm, no load balancer, no Frontdoor. Thank you,Beau D'AmoreOn Aug 30, 2023, at 1:11 AM, Joakim Westin ***@***.***> wrote:﻿
Just a follow up, where is the SSL connection terminated? (who manages the SSL connection?)
